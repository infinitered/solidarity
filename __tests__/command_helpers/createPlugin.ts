const createPlugin: Function = require('../../src/extensions/functions/createPlugin')
const context = require('mockContext')

test('check result shape', () => {
  expect(createPlugin).toMatchSnapshot()
})

test('investigate createPlugin', async () => {
  const result = await createPlugin(context)
  expect(result).toMatchSnapshot()
  expect(context.template.generate).toBeCalled()
  expect(context.prompt.ask).toBeCalled()
  expect(context.prompt.confirm).toBeCalled()
  expect(context.print.success).toBeCalled()
})

describe('checking plugin paths', () => {
  beforeEach(() => {
    const mockedPrompt = jest
      .fn()
      .mockImplementationOnce(() => Promise.resolve({ plugin: 'Nachos' }))
      .mockImplementationOnce(() => Promise.resolve({ pluginDesc: 'das nachos plugin' }))

    context.prompt.ask = mockedPrompt
  })

  test(`Choice 1 - 'I do not want a generated rule file'`, async () => {
    context.prompt.ask.mockImplementationOnce(() =>
      Promise.resolve({
        ruleChoice: 'I do not want a generated rule file',
      })
    )

    const result = await createPlugin(context)
    expect(result).toMatchSnapshot()
  })

  test(`Choice 2 - 'Just a simple rule template'`, async () => {
    context.prompt.ask.mockImplementationOnce(() =>
      Promise.resolve({
        ruleChoice: 'Just a simple rule template',
      })
    )

    const result = await createPlugin(context)
    expect(result).toMatchSnapshot()
  })

  test(`Choice 3 - 'Template + optional rules'`, async () => {
    context.prompt.ask.mockImplementationOnce(() =>
      Promise.resolve({
        ruleChoice: 'Template + optional rules',
      })
    )

    const result = await createPlugin(context)
    expect(result).toMatchSnapshot()
  })
})
