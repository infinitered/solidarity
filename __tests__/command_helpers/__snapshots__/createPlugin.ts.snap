// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check result shape 1`] = `[Function]`;

exports[`checking plugin paths Choice 1 - 'I do not want a generated rule file' 1`] = `
Array [
  Array [
    ".gitignore.ejs",
    ".gitignore",
  ],
  Array [
    "README.md.ejs",
    "README.md",
  ],
  Array [
    "package.json.ejs",
    "package.json",
  ],
  Array [
    "simple-plugin.js.ejs",
    "extensions/solidarity-nachos.js",
  ],
]
`;

exports[`checking plugin paths Choice 2 - 'Just a simple rule template' 1`] = `
Array [
  Array [
    ".gitignore.ejs",
    ".gitignore",
  ],
  Array [
    "README.md.ejs",
    "README.md",
  ],
  Array [
    "package.json.ejs",
    "package.json",
  ],
  Array [
    "rules-template.json.ejs",
    "templates/solidarity-nachos-template.json",
  ],
  Array [
    "simple-plugin.js.ejs",
    "extensions/solidarity-nachos.js",
  ],
]
`;

exports[`checking plugin paths Choice 3 - 'Template + optional rules' 1`] = `
Array [
  Array [
    ".gitignore.ejs",
    ".gitignore",
  ],
  Array [
    "README.md.ejs",
    "README.md",
  ],
  Array [
    "package.json.ejs",
    "package.json",
  ],
  Array [
    "rules-template.json.ejs",
    "templates/solidarity-nachos-template.json",
  ],
  Array [
    "helpful-plugin.js.ejs",
    "extensions/solidarity-nachos.js",
  ],
  Array [
    "addOptionalRules.js.ejs",
    "extensions/helpers/addOptionalRules.js",
  ],
]
`;

exports[`investigate createPlugin 1`] = `
Array [
  Array [
    ".gitignore.ejs",
    ".gitignore",
  ],
  Array [
    "README.md.ejs",
    "README.md",
  ],
  Array [
    "package.json.ejs",
    "package.json",
  ],
]
`;
